pkgs/development/python-modules/py-sr25519-bindings/default.nix

```diff
@@ -0,0 +1,58 @@
+{ lib
+, fetchFromGitHub
+, fetchpatch
+, buildPythonPackage
+, rustPlatform
+, stdenv
+, libiconv }:
+
+buildPythonPackage rec {
+  pname = "py-sr25519-bindings";
+  version = "0.1.5";
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "polkascan";
+    repo = "py-sr25519-bindings";
+    rev = "v${version}";
+    sha256 = "sha256-Pv6pmtA6y9vnGMH0CmWHwkwiU3HaqT3QD8vDjrX4INE=";
+  };
+
+  patches = [
+    ./cargo.lock.patch
+    (fetchpatch {
+      name = "update-pyo3-to-0.11.0.patch";
+      url = "https://github.com/polkascan/py-sr25519-bindings/commit/7703d229152d6f39d5a0ab5a9928dfe1239b9aad.patch";
+      hash = "sha256-HleuiUDBwBuPwjSBcMwnfPlkHjqjcEHDJvkoV+GkJ5s=";
+    })
+  ];
+
+  cargoDeps = rustPlatform.fetchCargoTarball {
+    inherit src;
+    name = "${pname}-${version}";
+    hash = "sha256-BiLwE42T8AdjpQG6htOlnDIp2Gu0ewbyo9iX/KH4weQ=";
+    patches = [
```

Comment:
Can you not factor out the `patches` here instead of repeating yourself?

Reply:
Patch has since been upstreamed.